tests/tests.go

```diff
@@ -907,17 +868,23 @@ func (ctl *controller) waitIssuePipelineImpl(id int, approveFunc func(issue *api
 		}
 		switch status {
 		case api.TaskPendingApproval:
+			if approveOnce && approve {
+				return api.TaskDone, nil
+			}
 			if err := approveFunc(issue); err != nil {
 				return api.TaskFailed, err
 			}
+			approve = true
 		case api.TaskFailed:
 			return status, err
 		case api.TaskDone:
 			return status, err
 		case api.TaskCanceled:
 			return status, err
 		case api.TaskPending:
+			approve = true
 		case api.TaskRunning:
+			approve = true
 			// no-op, keep waiting
```

Comment:
This comment is not appropriate now, please remove it.

Reply:
If the task is still running now, we will still wait.